test/lib/mutex/acquire.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const Utils = require('../../test_lib/utils.js');
+
+const MutexAddress = artifacts.require('TestMutexAddress');
+
+contract('MutexAddress.acquire()', async (accounts) => {
+
+  it('should acquire lock for an address', async () => {
+
+    let mutex = await MutexAddress.new();
+
+    let address = accounts[0];
+    let result = await mutex.acquireExternal.call(address);
+
+    assert.strictEqual(
+      result,
+      true,
+      'Lock acquire should succeed.'
+    );
+    await mutex.acquireExternal(address);
+
+    await Utils.expectRevert(
```

Comment:
We already have a separate test for this.

Reply:
This is to assert if the lock is acquired.

Reply:
Why no public locks mapping?